File: services/education/internal/model/campus_data_clear.go
package model

const CampusDataClearScopeBusinessOnly = "business_only"

type CampusDataClearRequest struct {
	Scope string `json:"scope"`
}

type CampusDataClearSummary struct {
	Students                   int `json:"students"`
	StudentFieldValues         int `json:"studentFieldValues"`
	StudentChangeRecords       int `json:"studentChangeRecords"`
	FaceProfiles               int `json:"faceProfiles"`
	FaceAttendanceSessions     int `json:"faceAttendanceSessions"`
	FaceAttendanceRecords      int `json:"faceAttendanceRecords"`
	FaceRollCallTasks          int `json:"faceRollCallTasks"`
	StudentTeachingRecords     int `json:"studentTeachingRecords"`
	StudentTeachingChangeLogs  int `json:"studentTeachingChangeLogs"`
	StudentRehabRecords        int `json:"studentRehabRecords"`
	FollowRecords              int `json:"followRecords"`
	HomeworkTasks              int `json:"homeworkTasks"`
	NoticeRecords              int `json:"noticeRecords"`
	Orders                     int `json:"orders"`
	OrderCourseDetails         int `json:"orderCourseDetails"`
	OrderPaymentDetails        int `json:"orderPaymentDetails"`
	Ledgers                    int `json:"ledgers"`
	ApprovalRecords            int `json:"approvalRecords"`
	ApprovalHistories          int `json:"approvalHistories"`
	TuitionAccounts            int `json:"tuitionAccounts"`
	TuitionAccountFlows        int `json:"tuitionAccountFlows"`
	CloseTuitionAccountOrders  int `json:"closeTuitionAccountOrders"`
	RefundTuitionOrders        int `json:"refundTuitionOrders"`
	RefundTuitionOrderItems    int `json:"refundTuitionOrderItems"`
	SuspendResumeTuitionOrders int `json:"suspendResumeTuitionOrders"`
	RechargeAccounts           int `json:"rechargeAccounts"`
	RechargeAccountStudents    int `json:"rechargeAccountStudents"`
	RechargeAccountFlows       int `json:"rechargeAccountFlows"`
	RechargeAccountOrders      int `json:"rechargeAccountOrders"`
	RechargeAccountOrderTags   int `json:"rechargeAccountOrderTags"`
	RechargeAccountBills       int `json:"rechargeAccountBills"`
	RechargeAccountBillFlows   int `json:"rechargeAccountBillFlows"`
	Courses                    int `json:"courses"`
	CourseDetails              int `json:"courseDetails"`
	CourseQuotations           int `json:"courseQuotations"`
	CoursePropertyResults      int `json:"coursePropertyResults"`
	ProductPackages            int `json:"productPackages"`
	ProductPackageItems        int `json:"productPackageItems"`
	ProductPackageProperties   int `json:"productPackageProperties"`
	CourseSaleVolumesReset     int `json:"courseSaleVolumesReset"`
	ImportTasks                int `json:"importTasks"`
	ImportTaskRecords          int `json:"importTaskRecords"`
	OrderImportTasks           int `json:"orderImportTasks"`
	OrderImportTaskRecords     int `json:"orderImportTaskRecords"`
	RechargeImportTasks        int `json:"rechargeImportTasks"`
	RechargeImportTaskRecords  int `json:"rechargeImportTaskRecords"`
	ExportRecords              int `json:"exportRecords"`
	TemplateMessageRecords     int `json:"templateMessageRecords"`
	TemplateMessageRecordItems int `json:"templateMessageRecordItems"`
	WeChatBindTickets          int `json:"weChatBindTickets"`
	WeChatStudentBindings      int `json:"weChatStudentBindings"`
	TeachingClasses            int `json:"teachingClasses"`
	TeachingClassStudents      int `json:"teachingClassStudents"`
	TeachingClassTeachers      int `json:"teachingClassTeachers"`
	TeachingSchedules          int `json:"teachingSchedules"`
	TeachingScheduleStudents   int `json:"teachingScheduleStudents"`
	TeachingScheduleBatchMetas int `json:"teachingScheduleBatchMetas"`
	TeachingRecords            int `json:"teachingRecords"`
	TeachingClassOperationLogs int `json:"teachingClassOperationLogs"`
	TeachingClassEntryExits    int `json:"teachingClassEntryExits"`
}

type CampusDataClearResult struct {
	Scope                     string                 `json:"scope"`
	ScopeName                 string                 `json:"scopeName"`
	Cleared                   CampusDataClearSummary `json:"cleared"`
	Preserved                 []string               `json:"preserved"`
	IntentStudentIndexCleared bool                   `json:"intentStudentIndexCleared"`
	IntentStudentIndexMessage string                 `json:"intentStudentIndexMessage,omitempty"`
}

File: institution-admin/src/api/business-settings/campus-data-clear.ts
import { usePost } from '~/utils/request'

export const CAMPUS_DATA_CLEAR_SCOPE_BUSINESS_ONLY = 'business_only'

export interface CampusDataClearResult {
  scope: string
  scopeName: string
  cleared: {
    students: number
    studentFieldValues: number
    studentChangeRecords: number
    studentTeachingRecords: number
    studentTeachingChangeLogs: number
    studentRehabRecords: number
    followRecords: number
    homeworkTasks: number
    noticeRecords: number
    orders: number
    orderCourseDetails: number
    orderPaymentDetails: number
    ledgers: number
    approvalRecords: number
    approvalHistories: number
    tuitionAccounts: number
    tuitionAccountFlows: number
    closeTuitionAccountOrders: number
    refundTuitionOrders: number
    refundTuitionOrderItems: number
    suspendResumeTuitionOrders: number
    rechargeAccounts: number
    rechargeAccountStudents: number
    rechargeAccountFlows: number
    rechargeAccountOrders: number
    rechargeAccountOrderTags: number
    rechargeAccountBills: number
    rechargeAccountBillFlows: number
    courses: number
    courseDetails: number
    courseQuotations: number
    coursePropertyResults: number
    productPackages: number
    productPackageItems: number
    productPackageProperties: number
    importTasks: number
    importTaskRecords: number
    orderImportTasks: number
    orderImportTaskRecords: number
    rechargeImportTasks: number
    rechargeImportTaskRecords: number
    exportRecords: number
    templateMessageRecords: number
    templateMessageRecordItems: number
    weChatBindTickets: number
    weChatStudentBindings: number
    courseSaleVolumesReset: number
    teachingClasses: number
    teachingClassStudents: number
    teachingClassTeachers: number
    teachingSchedules: number
    teachingScheduleStudents: number
    teachingScheduleBatchMetas: number
    teachingRecords: number
    teachingClassOperationLogs: number
    teachingClassEntryExits: number
  }
  preserved: string[]
  intentStudentIndexCleared: boolean
  intentStudentIndexMessage?: string
}

export interface CampusDataClearPayload {
  scope?: string
}

export function clearCampusDataApi(data: CampusDataClearPayload = { scope: CAMPUS_DATA_CLEAR_SCOPE_BUSINESS_ONLY }) {
  return usePost<CampusDataClearResult>('/api/v1/campus-data/clear', data)
}
